src/features/areaAssignments/store.ts

```diff
@@ -250,32 +255,55 @@ const areaAssignmentSlice = createSlice({
       const [assignmentId, location] = action.payload;
 
       remoteItemUpdated(state.locationsByAssignmentId[assignmentId], location);
+
+      Object.keys(state.locationsByAssignmentIdAndAreaId).forEach((key) => {
+        const [keyAssignmentIdStr] = key.split(':');
+        const keyAssignmentId = Number(keyAssignmentIdStr);
+
+        if (keyAssignmentId == assignmentId) {
+          const list = state.locationsByAssignmentIdAndAreaId[key];
+          if (list.items.some((item) => item.id == location.id)) {
+            remoteItemUpdated(list, location);
+          }
+        }
+      });
     },
     locationUpdated: (state, action: PayloadAction<ZetkinLocation>) => {
       const location = action.payload;
 
       Object.values(state.locationsByAssignmentId).forEach((list) => {
         remoteItemUpdated(list, location);
       });
+      Object.values(state.locationsByAssignmentIdAndAreaId).forEach((list) => {
+        remoteItemUpdated(list, location);
+      });
```

Comment:
I worry that overwriting the location object for all areas and assignments could overwrite stats from one assignment in the store for another. This is an edge case and is solved by refreshing the browser, so probably not a big deal. But it's something to think about and potentially revisit this for a refactor (and/or adding tests) down the line.